// sekigae/index.js
'use strict';

// load /lib/sekigae.js
var Sekigae = require('./lib/sekigae.js');

Sekigae.prototype.getText = function() {
  // set bot text
  return this.makeMainText() + '\n' + this.makeDayText();
};

Sekigae.prototype.makeMainText = function() {
  if (this.checkHoliday()) {
    return 'It is holiday!!';
  } else {
    var memberList = this.getYaml(this.yamlName);
    var randomArray = this.getRandomArray(15);

    var mainText = '';
    var tmp = this.setMemberDeskHash(memberList, randomArray);

    for(var key in tmp) {
        mainText += key + ': ' + tmp[key] + '\n';
    };
    return mainText;
  };
};

Sekigae.prototype.makeDayText = function() {
    return '今日は' + this.today + 'です。';
};

Sekigae.prototype.sendPostRequest = function(url, text) {
  // set request module
  var request = require('request');
  // set url, text and so on...
  var options = {
    url: url,
    method: 'POST',
    headers: { 'Content-Type' : 'application/json' },
    json: true,
    body: { 'text' : text }
  };

  // send POST request & display response
  request(options, function(error, response, body) {
    if (!error && response.statusCode == 200) {
      console.log(body.name);
    } else {
      console.log('error: ' + response.statusCode)
    };
  });
}

// export js files
module.exports = Sekigae;
