// sekigae/lib/sekigae.js
'use strict';

//
// sample codes START
//

// console.log('/sekigae/lib/sekigae.js is loaded') // WORK

// function Sekigae() {
//     this.my_func1 = function() {
//         return 'piyopiyo';
//     };
// };

// Sekigae.prototype.my_func2 = function() {
//     return 'hogehoge';
// };

// function my_func3() {
//     return 'fugafuga pya-';
// };

// module.exports = Sekigae;
// module.exports.my_func3 = my_func3();

//
// sample codes END
//

function Sekigae() {};

Sekigae.prototype.getToday = function(difference) {
    let date = new Date();
    // add difference between server & your local TimeZone
    date.setHours(date.getHours() + difference);

    let year = date.getFullYear();

    // add 1 (this method returns 0 to 11)
    let month = date.getMonth() + 1;
    // fix format
    month = (month < 10 ? '0' : '') + month;

    let day = date.getDate();
    // fix format
    day = (day < 10 ? '0' : '') + day;

    // get Day Name as number
    let weeknum = date.getDay();

    // set Day Name list
    let dayName = ['日曜日', '月曜日', '火曜日', '水曜日', '木曜日', '金曜日', '土曜日']

    return year + '/' + month + '/' + day + ' ' + dayName[weeknum]
};

Sekigae.prototype.getYaml = function(filename) {
    var yaml = require('js-yaml');
    var fs = require('fs');

    try {
        let doc = yaml.safeLoad(fs.readFileSync(`./${filename}`, 'utf-8'));
        return doc;
    } catch (e) {
        console.log(e);
    };
};

module.exports = Sekigae;
